test/test_um2netcdf.py

```diff
@@ -5,7 +5,209 @@
+                     12499.999090756188, 13520.000611878331, 14580.799681536007,
+                     15694.639882321579, 16875.311437270288, 18138.62619334655,
+                     19503.01036943094, 20990.18759042441, 22626.081748420565,
+                     24458.285403646936, 26583.640230535515, 29219.080215877355,
+                     32908.69305496925, 39254.833576], dtype=object)
+    return data
+
+
+@pytest.fixture
+def mule_vars(z_sea_rho_data, z_sea_theta_data):
+    """Simulate mule variables from aiihca.paa1jan.subset data."""
+    d_lat = 1.25  # spacing manually copied from aiihca.paa1jan.subset file
+    d_lon = 1.875
+    return um2nc.MuleVars(um2nc.GRID_NEW_DYNAMICS, d_lat, d_lon, z_sea_rho_data, z_sea_theta_data)
+
+
+@pytest.fixture
+def air_temp_cube():
+    # copied from aiihca.paa1jan.subset file
+    return DummyCube(30204, "air_temperature")
+
+
+@pytest.fixture
+def precipitation_flux_cube():
+    # copied from aiihca.paa1jan.subset file
+    return DummyCube(5216, "precipitation_flux")
+
+
+@pytest.fixture
+def std_args():
+    # TODO: make args namedtuple?
+    args = mock.Mock()
+    args.nomask = False
+    args.nohist = False
+    args.nckind = 3
+    args.include_list = None
+    args.exclude_list = None
+    args.simple = False
+    args.verbose = False
+    return args
+
+
+def test_process_without_masking(air_temp_cube, precipitation_flux_cube, mule_vars, std_args):
+    """Attempts end-to-end test of process(), ignoring cubes requiring masking."""
+
+    # FIXME: this convoluted setup is a big code stench
+    #        use this to gradually refactor process()
+    #        add naming vars to the dummy cubes
+    with mock.patch("mule.load_umfile"):  # ignore m_load_umfile as process_mule_vars is mocked
+        with mock.patch("umpost.um2netcdf.process_mule_vars") as m_mule_vars:
+            m_mule_vars.return_value = mule_vars
+
+            with mock.patch("iris.load") as m_iris_load:
+                cubes = [air_temp_cube, precipitation_flux_cube]
+
+                for c in cubes:
+                    c.attributes = {um2nc.STASH: DummyStash(*split_item_code(c.item_code))}
+                    c.cell_methods = []
+
+                    # TODO: replace with DummyCubeWithCoords ()?
+                    #       or add coord fixtures?
+                    c.coord["latitude"] = 0.0  # FIXME
+                    c.coord["longitude"] = 0.0  # FIXME
+
+                m_iris_load.return_value = cubes
+
+                with mock.patch("iris.fileformats.netcdf.Saver") as m_saver:  # prevent I/O
+                    # mock `sman` var to prevent I/O
+                    m_sman = mock.Mock()
+                    m_saver().__enter__.return_value = m_sman
+
+                    # TODO: fix lat/lon & levels requires c.coord attribute
+                    #       use fixtures to add attrs & remove the patches?
+                    with mock.patch("umpost.um2netcdf.fix_latlon_coord") as m_coord:
```

Comment:
Here's an example of nested `mock.patch()` fun. These are _pseudo practical_ mocks to ignore code which requires a `c.coord` data attribute. Spencer is working on a test data structure with `coords`, which could potentially fit here & reduce the number of patches.

Reply:
@blimlim's work on #63 adds a `DummyCoordinate` data class to simulate `coord` attrs. When #63 is merged, I'll refactor the `test_process_...()` test setup to remove the `coord` mock & ensure the prod code is called.

Removing the `fix_level_coord()` & `cubewrite()` mocks can be handled in separate PRs (working on a model of gradually improving tests as `um2nc` is improved).

Reply:
This is covered in #69.